core/src/test/java/org/apache/calcite/test/UdfTest.java

```diff
@@ -407,14 +414,33 @@ private CalciteAssert.AssertThat withUdf() {
         .returns("P=null\n");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6645">[CALCITE-6645]
+   * User-defined function with niladic parentheses</a>. */
+  @Test void testUserDefinedFunctionWithNiladicParentheses() {
```

Comment:
This looks pretty good.
Are there already negative tests that show errors when the conformance is wrong?

Reply:
Thanks for your suggestion, I have added some negative tests.

Reply:
In general you should wait to squash the commits until the PR has been approved.
This makes reviewing changes easier.

Reply:
Thank you for your guidance. I will pay attention to this point in my future PRs.